typings/pinejs-client-core.d.ts

```diff
@@ -379,10 +388,11 @@ export type ODataOptionsStrict<T> = Omit<
 export type ODataOptionsWithFilter<T> = ODataOptions<T> &
 	Required<Pick<ODataOptions<T>, '$filter'>>;
 
+export type ReplaceWebResource<K> = K extends WebResource ? WebResourceFile : K;
 export type SubmitBody<T> = {
 	[k in keyof T]?: T[k] extends AssociatedResource<object>
 		? number | null
-		: T[k];
+		: ReplaceWebResource<T[k]>;
```

Comment:
Unless you need to use this elsewhere, I would just inline the `ReplaceWebResource`